lib/liquid/template.rb

```diff
@@ -254,7 +254,7 @@ def with_profiling(context)
       if @profiling && !context.partial
         raise "Profiler not loaded, require 'liquid/profiler' first" unless defined?(Liquid::Profiler)
 
-        @profiler = Profiler.new
+        @profiler = Profiler.new(context.template_name)
```

Comment:
Just noting that this changes normal behavior. Anyone relying on `<root>` detection may experience issues. I don't expect this to impact anyone.